gammapy/modeling/fit.py

```diff
@@ -359,23 +359,63 @@ def stat_profile(
 
         return {"values": values, "stat": np.array(stats)}
 
-    def stat_contour(self):
+    def stat_surface(
+        self, x, y, x_values, y_values, reoptimize=False, **optimize_opts
+    ):
         """Compute fit statistic contour.
 
         The method used is to vary two parameters, keeping all others fixed.
         So this is taking a "slice" or "scan" of the fit statistic.
 
+        Caveat: This method can be very computationally intensive and slow. In future releases, parallel computation
+        will make it more efficient.
+
         See also: `Fit.minos_contour`
 
         Parameters
         ----------
-        TODO
+        x, y : `~gammapy.modeling.Parameter`
+            Parameters of interest
+        x_values, y_values : list or `numpy.ndarray`
+            Parameter values to evaluate the fit statistic for.
+        reoptimize : bool
+            Re-optimize other parameters, when computing the fit statistic profile.
+        **optimize_opts : dict
+            Keyword arguments passed to the optimizer. See `Fit.optimize` for further details.
 
         Returns
         -------
-        TODO
+        results : dict
+            Dictionary with keys "x_values", "y_values" and "stat".
+
         """
-        raise NotImplementedError
+        parameters = self._parameters
+        x = parameters[x]
+        y = parameters[y]
+
+        stats = []
+        with parameters.restore_values:
+            for x_value, y_value in itertools.product(x_values, y_values):
+                # TODO: Remove log.info() and provide a nice progress bar
+                log.info(f"Processing: x={x_value}, y={y_value}")
+                x.value = x_value
+                y.value = y_value
+                if reoptimize:
+                    x.frozen = True
+                    y.frozen = True
+                    result = self.optimize(**optimize_opts)
+                    stat = result.total_stat
+                else:
+                    stat = self.datasets.stat_sum()
+
+                stats.append(stat)
+
+        stats = np.array(stats)
+        stats = stats.reshape(
+            (np.asarray(x_values).shape[0], np.asarray(y_values).shape[0])
+        )
+
+        return {"x_values": x_values, "y_values": y_values, "stat": stats}
```

Comment:
For consistency with `Estimators`, I would suggest to replace `stat` by `stat_scan` in the keys of the returned dict and possibly `x_values` by `x_scan`.
Could we use the actual parameter name rather than `x`?

Reply:
Same comment as above. I used `stat` and `x/y` because it is consistent with `Fit.stat_profile` and`Fit.minos_contour()`. I would suggest to merge this, and later clean up the code to ensure consistency between `Fit` methods and `Estimator` classes